### PR TITLE
Add Beneficiary model

### PR DESCRIPTION
**Pull request by jmnda-dev**
- [x] add Beneficiary model
- [x] register Beneficiary model in admin

### CODE REVIEW

File: itez/beneficiary/admin.py [hunk 1]
```diff
@@ -3,27 +3,38 @@
...
-        'insured',
+        "gross_pay",
+        "company",
+        "work_address",
+        "insured",
     ]
 
+
+class BeneficiaryAdmin(admin.ModelAdmin):
+    list_display = ["first_name", "last_name", "beneficiary_code", "created"]
```
Comment (sonlinux) [inline itez/beneficiary/admin.py]: lets mash this top to bottom and left shift 
reference to `WorkdetailAdmin()`

File: itez/beneficiary/migrations/0008_beneficiary.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Generated by Django 3.1.13 on 2021-11-04 09:29
+
+from django.db import migrations, models
+import uuid
```
Comment (sonlinux) [inline itez/beneficiary/migrations/0008_beneficiary.py:4]: built in python imports should always come first before django imports

File: itez/beneficiary/models.py [hunk 3]
```diff
@@ -1,38 +1,72 @@
 from django.db import models
-from django.db.models.deletion import PROTECT 
+from django.db.models.deletion import PROTECT
+from django.db.models.fields.related import create_many_to_many_intermediary_model
 from django.utils.translation import gettext_lazy as _
+from .utils import generate_uuid_and_agent_code
```
Comment (sonlinux) [inline itez/beneficiary/models.py:5]: seperate python, djang, local imports with a single empty line space

File: itez/beneficiary/models.py [hunk 4]
```diff
@@ -1,38 +1,72 @@
...
+
+    MARITAL_STATUS = (
+        ("single", _("Single")),
+        ("married", _("Married")),
+        ("seperated", _("Seperated")),
+        ("divorced", _("Divorced")),
+        ("widowed", _("Widowed")),
+    )
+
+    GENDER = (("male", _("Male")), ("female", _("Female")), ("other", _("Other")))
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: shift this top to bottom like `MARITAL_STATUS`

File: itez/beneficiary/models.py [hunk 5]
```diff
@@ -1,38 +1,72 @@
...
+
+    first_name = models.CharField(_("First Name"), max_length=200)
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: check on existing models 
object data should be top to bottom and with a left shift

File: itez/beneficiary/models.py [hunk 6]
```diff
@@ -1,38 +1,72 @@
...
+    last_name = models.CharField(_("Last Name"), max_length=200)
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here

File: itez/beneficiary/models.py [hunk 7]
```diff
@@ -1,38 +1,72 @@
...
+    other_name = models.CharField(_("Other Name"), max_length=200, null=True)
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here

File: itez/beneficiary/models.py [hunk 8]
```diff
@@ -1,38 +1,72 @@
...
+    gender = models.CharField(_("Gender"), max_length=100, choices=GENDER)
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here

File: itez/beneficiary/models.py [hunk 9]
```diff
@@ -1,38 +1,72 @@
...
+    profile_photo = models.ImageField(upload_to="profile_photo/")
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here

File: itez/beneficiary/models.py [hunk 10]
```diff
@@ -1,38 +1,72 @@
...
+    beneficiary_code = models.UUIDField(
+        default=generate_uuid_and_agent_code()[0], editable=False
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: great, but  editable shifts to  the left Remember top to bottom

File: itez/beneficiary/models.py [hunk 11]
```diff
@@ -1,38 +1,72 @@
...
+    )
+    date_of_birth = models.DateField(_("Date of Birth"))
```
Comment (sonlinux) [inline itez/beneficiary/models.py:78]: same here

File: itez/beneficiary/models.py [hunk 12]
```diff
@@ -1,38 +1,72 @@
...
+    marital_status = models.CharField(
```
Comment (sonlinux) [inline itez/beneficiary/models.py:79]: same here

File: itez/beneficiary/models.py [hunk 13]
```diff
@@ -1,38 +1,72 @@
...
+        _("Marital Status"), choices=MARITAL_STATUS, max_length=100
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here left shift

File: itez/beneficiary/models.py [hunk 14]
```diff
@@ -1,38 +1,72 @@
...
 class Province(models.Model):
     """
     Implements province properties and required methods.
     """
-    name = models.CharField(
-        _("Province"),
-        max_length=255 
-    )
+
+    name = models.CharField(_("Province"), max_length=255)
```
Comment (sonlinux) [inline itez/beneficiary/models.py:191]: left shift and top to bottom

File: itez/beneficiary/models.py [hunk 15]
```diff
@@ -1,38 +1,72 @@
...
     """
     Define district properties and corresponding methods.
     """
-    
-    name = models.CharField(
-        _('District'),
-        max_length=255
-    )
+
+    name = models.CharField(_("District"), max_length=255)
```
Comment (sonlinux) [inline itez/beneficiary/models.py:209]: same here

File: itez/beneficiary/models.py [hunk 16]
```diff
@@ -49,15 +83,10 @@ class ServiceArea(models.Model):
...
-    name = models.CharField(
-        _('Service Area'),
-        max_length=200
-    )
 
-    district = models.ForeignKey(
-        District,
-        on_delete=PROTECT
-    )
+    name = models.CharField(_("Service Area"), max_length=200)
```
Comment (sonlinux) [inline itez/beneficiary/models.py:227]: service area model now exists and we dont need it on beneficiary
the only linkage we need is district and service area (this one can be null and blank) but district should be there

we can get province via district and or service area if provided

File: itez/beneficiary/models.py [hunk 17]
```diff
@@ -67,28 +96,15 @@ def __str__(self):
...
-    insured = models.BooleanField(
-        _("Company Insured"),
-        default = False
-    )
-    work_address = models.TextField(
-        _("Work Address"),
-        null=True,
-        blank=True
+
+    gross_pay = models.DecimalField(
```
Comment (sonlinux) [inline itez/beneficiary/models.py:242]: left shift top to bottom

File: itez/beneficiary/models.py [hunk 18]
```diff
@@ -67,28 +96,15 @@ def __str__(self):
...
+        _("Monthly Salary"), decimal_places=2, max_digits=1000, null=False
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: same here

File: itez/beneficiary/models.py [hunk 19]
```diff
@@ -67,28 +96,15 @@ def __str__(self):
...
     )
+    company = models.CharField(_("Company Name"), max_length=200, null=False)
+    insured = models.BooleanField(_("Company Insured"), default=False)
+    work_address = models.TextField(_("Work Address"), null=True, blank=True)
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: am wondering why apply your IDE changes when we are working against specific code formatting
change these too

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/migrations/0008_beneficiary.py (hunk 2 above)
Comment (jmnda-dev) [reply itez/beneficiary/migrations/0008_beneficiary.py:4]: This is in the migration file that was generated, I have no control over how imports are done

File: itez/beneficiary/models.py (hunk 19 above)
Comment (jmnda-dev) [reply itez/beneficiary/models.py]: My IDE is set to auto-format using black, I have disabled it for now.

**Review by sonlinux — APPROVED** [review]
LGTM

File: itez/beneficiary/models.py [hunk 20]
```diff
@@ -1,38 +1,96 @@
 from django.db import models
-from django.db.models.deletion import PROTECT 
+from django.db.models.deletion import PROTECT
+from django.db.models.fields.related import create_many_to_many_intermediary_model
```
Comment (sonlinux) [inline itez/beneficiary/models.py:2]: We do not need this

File: itez/beneficiary/models.py [hunk 21]
```diff
@@ -1,38 +1,96 @@
...
+    )
+    last_name = models.CharField(
+        _("Last Name"),
+        max_length=200
+    )
+    other_name = models.CharField(
+        _("Other Name"),
+        max_length=200,
+        null=True
+    )
```
Comment (sonlinux) [inline itez/beneficiary/models.py:53]: Add blank whenever you allow null

File: itez/beneficiary/models.py [hunk 22]
```diff
@@ -1,38 +1,96 @@
...
+    gender = models.CharField(
+        _("Gender"),
+        max_length=100,
+        choices=GENDER
+    )
+    profile_photo = models.ImageField(
```
Comment (sonlinux) [inline itez/beneficiary/models.py:59]: Allow this to be null and blank too

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/models.py [hunk 23]
```diff
@@ -1,42 +1,193 @@
...
+
+    MARITAL_STATUS = (
+        ("single", _("Single")),
+        ("married", _("Married")),
+        ("seperated", _("Seperated")),
+        ("divorced", _("Divorced")),
+        ("widowed", _("Widowed")),
+    )
+
+    TERTIARY_EDUCATION_QUALIFICATION = (
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: we also want to cover basic primary and secondary O'Leve education optoins and NONE for no formal school

File: itez/beneficiary/models.py [hunk 24]
```diff
@@ -1,42 +1,193 @@
...
+        max_length=20,
+        null=True,
+        blank=True
+    )
+    email = models.EmailField(
+        max_length=200,
+        null=True,
+        blank=True
+    )
+    beneficiary_code = models.UUIDField(
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: beneficiary_id

File: itez/beneficiary/models.py [hunk 25]
```diff
@@ -1,42 +1,193 @@
...
+    junior_secondary_qualification = models.CharField(
+        max_length=200,
+        null=True,
+        blank=True
+    )
+    senior_secondary_qualification = models.CharField(
+        max_length=200,
+        null=True,
+        blank=True,
+    )
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: these and teritery should be merged into `education_level` option field

File: itez/beneficiary/models.py [hunk 26]
```diff
@@ -1,42 +1,193 @@
...
+    tertiary_eduction_qualification = models.CharField(
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: `education_level`

File: itez/beneficiary/models.py [hunk 27]
```diff
@@ -1,42 +1,193 @@
...
+    def __str__(self):
+        return f"{self.first_name} {self.last_name}"
+
+    class Meta:
+        verbose_name = "Beneficiary"
+        verbose_name_plural = "Beneficiaries"
+        ordering = ["created"]
+
+
+class BeneficiaryParent(models.Model):
```
Comment (sonlinux) [inline itez/beneficiary/models.py:124]: we can get dobs for parents too and their villages

File: itez/beneficiary/models.py [hunk 28]
```diff
@@ -1,42 +1,193 @@
...
+    )
+    phone_number = models.CharField(
+        max_length=20,
+        null=True,
+        blank=True
+    )
+    created = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self) :
+        return f"Beneficiary Parent: {self.father_first_name} {self.father_last_name}"
```
Comment (sonlinux) [inline itez/beneficiary/models.py]:
```
 return f"Beneficiary Parents are \
            Father: {self.father_first_name} {self.father_last_name}, \
            Mother: {self.mother_first_name} {self.mother_last_name}"```

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/models.py (hunk 27 above)
Comment (jmnda-dev) [reply itez/beneficiary/models.py:124]: Added

File: itez/beneficiary/models.py (hunk 26 above)
Comment (jmnda-dev) [reply itez/beneficiary/models.py]: Done

File: itez/beneficiary/models.py [hunk 29]
```diff
@@ -24,7 +24,11 @@ class Beneficiary(models.Model):
         ("widowed", _("Widowed")),
     )
 
-    TERTIARY_EDUCATION_QUALIFICATION = (
+    EDUCATION_LEVEL = (
+        ("none", _("NONE")),
```
Comment (sonlinux) [inline itez/beneficiary/models.py]: None I meant :-)

**Review by sonlinux — APPROVED** [review]

File: itez/beneficiary/models.py (hunk 29 above)
Comment (sonlinux) [reply itez/beneficiary/models.py]: Just that and its a LGTM for a Squash and Merge

**Pull request merged**